rotkehlchen/data_import/importers/bitstamp.py

```diff
@@ -53,82 +56,42 @@ def _consume_bitstamp_transaction(
+            value_str, value_symbol = csv_row['Value'].split(' ')
+            fee_str, fee_symbol = csv_row['Fee'].split(' ')
+            spend_asset, spend_amount, receive_asset, receive_amount = get_swap_spend_receive(
+                raw_trade_type=csv_row['Sub Type'],
+                base_asset=asset_from_bitstamp(amount_symbol),
+                quote_asset=asset_from_bitstamp(value_symbol),
+                amount=(swap_amount := deserialize_asset_amount(amount_str)),
+                rate=Price(deserialize_asset_amount(value_str) / swap_amount),
             )
-            fee_event = HistoryEvent(
-                event_identifier=event_identifier,
-                sequence_index=2,
-                timestamp=timestamp,
-                location=Location.BITSTAMP,
-                asset=fee_currency,
-                amount=fee_amount,
-                notes=f'Fee of {fee_amount} {fee_currency} as the result of a trade on Bitstamp',
-                event_type=HistoryEventType.TRADE,
-                event_subtype=HistoryEventSubType.FEE,
+            self.add_history_events(
+                write_cursor=write_cursor,
+                history_events=create_swap_events(
+                    timestamp=timestamp,
+                    spend_asset=spend_asset,
+                    spend_amount=spend_amount,
+                    location=Location.BITSTAMP,
+                    receive_asset=receive_asset,
+                    receive_amount=receive_amount,
+                    event_identifier=event_identifier,
+                    fee_amount=deserialize_fee(fee_str),
+                    fee_asset=asset_from_bitstamp(fee_symbol),
+                ),
             )
-            self.add_history_events(write_cursor, [
-                spend_trade_event,
-                receive_trade_event,
-                fee_event,
-            ])
         elif csv_row['Type'] in {'Deposit', 'Withdrawal'}:
-            amount = deserialize_asset_amount(amount)
-            asset = asset_from_bitstamp(amount_symbol)
-            transaction_type = csv_row['Type']
-            if transaction_type == 'Deposit':
-                event_type = HistoryEventType.DEPOSIT
-                event_subtype = HistoryEventSubType.DEPOSIT_ASSET
-            else:
-                event_type = HistoryEventType.WITHDRAWAL
-                event_subtype = HistoryEventSubType.REMOVE_ASSET
-            movement_event = HistoryEvent(
-                event_identifier=event_identifier,
-                sequence_index=0,
-                timestamp=timestamp,
-                location=Location.BITSTAMP,
-                asset=asset,
-                amount=amount,
-                notes=f'{transaction_type} of {amount} {asset} on Bitstamp',
-                event_type=event_type,
-                event_subtype=event_subtype,
+            self.add_history_events(
+                write_cursor=write_cursor,
+                history_events=[AssetMovement(
```

Comment:
```suggestion
            events = [AssetMovement(
```
Would probably be better to only have `add_history_events` once at the end when `events is not None` or something like that.